cypress/integration/SidePanelLabs.spec.ts

```diff
@@ -0,0 +1,82 @@
+import * as h from '../helpers';
+
+describe('Side Panel', () => {
+  before(() => {
+    h.stories.visit();
+  });
+
+  ['Default', 'AsAside', 'AsDiv'].forEach(story => {
+    context(`given the "${story}" story is rendered`, () => {
+      beforeEach(() => {
+        h.stories.load('Testing|React/Labs/Side Panel/Cypress', story);
+      });
+
+      const name = /Accessible Label Name/i;
+
+      it('should not have any axe errors', () => {
+        cy.checkA11y();
+      });
+
+      context('the button', () => {
+        it(`should have an accessible name`, () => {
+          cy.findByRole('button', {name}).should('exist');
+        });
+
+        it(`should have an aria-expanded attribute of 'true'`, () => {
+          cy.findByRole('button', {name}).should('have.attr', 'aria-expanded', 'true');
+        });
+
+        it(`should have an aria-controls attribute equal to the id of the panel`, () => {
+          cy.findByRole('button', {name}).then(button => {
+            const buttonAriaControlsValue = button.attr('aria-controls');
+            switch (story) {
+              case 'AsAside':
+                cy.findByRole('complementary', {name}).should(
+                  'have.attr',
+                  'id',
+                  buttonAriaControlsValue
+                );
+                break;
+              default:
+                cy.findByRole('region', {name}).should('have.attr', 'id', buttonAriaControlsValue);
+            }
+          });
+        });
+      });
```

Comment:
Really not a fan of logic in test files, especially `switch` statements. I've done this type of thing before in the `forEach` array instead:
```ts
[
  { story: 'Default', role: 'region' },
  { story: 'AsAside', role: 'complementary' },
  { story: 'AsDiv', role: 'region' }
].forEach(({ story, role }) => {
  // story is the same as before
  // role is now the role you expect from the story
})
```

Another way to think about it is you create a story with your accessibility specifications. Another story can be used for a specification that applies to a different configuration, but not all specifications have to be run on every variant - especially if that requires `if` or `switch` statements.